File: app/javascript/packs/fee_structure.js

function updateTotal() {
  var admissionFees = parseFloat(document.getElementById('admission-fees').value) || 0;
  var annualAdmissionFees = parseFloat(document.getElementById('annual-admission-fees').value) || 0;
  var cautionMoney = parseFloat(document.getElementById('caution-money').value) || 0;
  var quarterlyTuitionFees = parseFloat(document.getElementById('quarterly-tuition-fees').value) || 0;
  var idCardFees = parseFloat(document.getElementById('id-card-fees').value) || 0;

  var total = admissionFees + annualAdmissionFees + cautionMoney + quarterlyTuitionFees + idCardFees;

  document.getElementById('total-fees').value = total;
}

document.getElementById('admission-fees').addEventListener('keyup', updateTotal);
document.getElementById('annual-admission-fees').addEventListener('keyup', updateTotal);
document.getElementById('caution-money').addEventListener('keyup', updateTotal);
document.getElementById('quarterly-tuition-fees').addEventListener('keyup', updateTotal);
document.getElementById('id-card-fees').addEventListener('keyup', updateTotal);

File: app/models/fee_structure.rb
# frozen_string_literal: true

# This is fee model
class FeeStructure < ApplicationRecord
  belongs_to :class_category

  validates :admission_fees, presence: { message: "admission fee can't be empty" },
                             numericality: { greater_than_or_equal_to: 0, message: 'must be non-negative' }
  validates :annual_admission_fees, presence: { message: "annual admission fee can't be empty" },
                                    numericality: { greater_than_or_equal_to: 0, message: 'must be non-negative' }
  validates :caution_money, presence: { message: "caution fee can't be empty" },
                            numericality: { greater_than_or_equal_to: 0, message: 'must be non-negative' }
  validates :quarterly_tuition_fees, presence: { message: "quarterly fee can't be empty" },
                                     numericality: { greater_than_or_equal_to: 0, message: 'must be non-negative' }
  validates :id_card_fees, presence: { message: "id card can't be empty" },
                           numericality: { greater_than_or_equal_to: 0, message: 'must be non-negative' }
  validates :total, presence: true, numericality: { greater_than_or_equal_to: 0 }
end
